src/backend/auth/adapter.ts

```diff
@@ -0,0 +1,195 @@
+import { Adapter, AdapterSession, AdapterUser } from "next-auth/adapters";
+import {
+  KyselyDatabaseInstance,
+  KyselySchema,
+  ZapatosTableNameToKyselySchema,
+} from "../lib";
+import { InsertObject, Selectable } from "kysely";
+
+const mapStoredUserToAdapterUser = (
+  user: Selectable<ZapatosTableNameToKyselySchema<"user">>
+): AdapterUser => ({
+  id: user.user_id,
+  email: user.email,
+  emailVerified: user.email_verified_at,
+  name: `${user.first_name} ${user.last_name}`,
+  image: user.image_url,
+});
+
+const mapStoredSessionToAdapterSession = (
+  session: Selectable<ZapatosTableNameToKyselySchema<"session">>
+): AdapterSession => ({
+  sessionToken: session.session_token,
+  userId: session.user_id,
+  expires: session.expires_at,
+});
+
+export const createPersistedAuthAdapter = (
+  db: KyselyDatabaseInstance
+): Adapter => ({
+  async createUser(user) {
+    const [first_name, last_name] = user.name?.split(" ") ?? [
```

Comment:
nit: splitting on space doesn't always work, for e.g. when someone's name has more than two parts.

Reply:
yeah, this is more of a best-effort

ideally users will be able to edit their name somewhere in the app